test/jit/test_cuda.py

```diff
@@ -0,0 +1,426 @@
+            # has been set to the default stream on d1
+            with torch.jit.cuda.device(d):
+                s3 = torch.cuda.current_stream(1)
+                check_s3 = s3.id() == s1.id()
+                check_d1 = torch.cuda._current_device() == s3.device_index()
+
+            # Check if the current device was reset to 0
+            is_device_d0 = torch.cuda._current_device() == s2.device_index()
+
+            return s0.device_index(), s1.device_index(), check_s2, check_s3, check_d0, check_d1, is_device_d0
+
+        d0, d1, check_s2, check_s3, check_d0, check_d1, is_device_d0 = fn()
+
+        self.assertEqual(d0, 0)
+        self.assertEqual(d1, 1)
+        self.assertTrue(check_s2)
+        self.assertTrue(check_s3)
+        self.assertTrue(check_d0)
+        self.assertTrue(check_d1)
+        self.assertTrue(is_device_d0)
+
+    @skipIfRocm
+    @unittest.skipIf(not TEST_MULTIGPU, "detected only one GPU")
+    @skipCUDANonDefaultStreamIf(True)
+    def test_streams(self):
+        # Check if a CUDA JIT stream is created
+        # on the _current_device
+        @torch.jit.script
+        def test_simple_stream():
+            device_index = torch.cuda._current_device()
+            s = torch.jit.cuda.Stream(device_index, 0)
+            return device_index == s.device_index()
+
+        self.assertTrue(test_simple_stream(), "Could not create Stream!")
+
+        # The test aims at checking different stream proporties.
+        class test_get_stream(torch.nn.Module):
+            def forward(self):
+                device_index = torch.cuda._current_device()
+                current_stream = torch.cuda.current_stream(device_index)
+                default_stream = torch.cuda.default_stream(device_index)
+                user_stream = torch.jit.cuda.Stream(device_index, 0)
+
+                # Check if the current and default streams are the same on the device
+                is_current_and_default_stream_same = current_stream.id() == default_stream.id()
+                # Check if user stream and default stream are not the same on the device
+                is_default_and_user_stream_not_same = default_stream.id() != user_stream.id()
+
+                with torch.jit.cuda.stream(user_stream):
+                    is_stream_set = torch.cuda.current_stream(device_index).id() == user_stream.id()
+
+                # Check if the stream was reset to current_stream
+                is_stream_reset = torch.cuda.current_stream(device_index).id() == current_stream.id()
+
+                tensor1 = torch.rand(10000, 10000, device = "cuda")
+                tensor2 = torch.mm(tensor1, tensor1).to("cuda")
+                default_stream.synchronize()
+                default_stream_query = default_stream.query()
+
+                return tensor1, tensor2, is_current_and_default_stream_same, is_default_and_user_stream_not_same, \
+                is_stream_set, is_stream_reset, default_stream_query, default_stream.id(), user_stream.id()
```

Comment:
Consider packing all these results into a TorchScript class and returning that. I think it would be more readable that way.

Reply:
Ok, so just a class to capture the return types?

Reply:
Ok done.